src/lib/anthropic-compat.ts

```diff
@@ -86,32 +77,56 @@ export function fromClaudeMessages(
   for (const msg of messages) {
     const { role, content } = msg;
 
-    // Handle string content directly
     if (typeof content === "string") {
       result.push(createEasyInputMessage(role, content));
       continue;
     }
 
-    // Handle array content - extract text and handle tool results
     const textParts: string[] = [];
+
     for (const block of content) {
-      if (isTextBlock(block)) {
-        textParts.push(block.text);
-      } else if (block.type === "tool_result") {
-        // Tool results need special handling - convert to function_call_output
-        const toolContent =
-          typeof block.content === "string"
-            ? block.content
-            : block.content.filter(isTextBlock).map((b) => b.text).join("");
-        result.push(createFunctionCallOutput(block.tool_use_id, toolContent));
+      switch (block.type) {
+        case 'text': {
+          const textBlock = block as models.ClaudeTextBlockParam;
+          textParts.push(textBlock.text);
+          // Note: cache_control is lost in conversion (OpenRouter doesn't support it)
+          break;
+        }
+
+        case 'image': {
+          // Images in input cannot be mapped to OpenRouter easy format
+          // Add text marker to preserve conversation flow
+          textParts.push('[Image content - not supported in OpenRouter format]');
+          break;
+        }
+
+        case 'tool_use': {
+          // Tool use blocks in input are conversation history, skip
```

Comment:
needs a dev mode warning that data is being intentionally thrown away
or change the name `fromClaudeMessages` to something that doesn't seem to imply that it is a generic message format transform